src/r_draw.c

```diff
@@ -526,21 +526,181 @@ static void R_DrawFuzzColumn_block(void)
     }
 }
 
-// [FG] spectre drawing mode: 0 original, 1 blocky (hires)
+#define FUZZDARK 6 * 256
 
-boolean fuzzcolumn_mode;
-void (*R_DrawFuzzColumn)(void) = R_DrawFuzzColumn_orig;
+static const int fuzzdark[FUZZTABLE] =
+{
+    FUZZDARK,0,FUZZDARK,0,FUZZDARK,FUZZDARK,0,
+    FUZZDARK,FUZZDARK,0,FUZZDARK,FUZZDARK,FUZZDARK,0,
+    FUZZDARK,FUZZDARK,FUZZDARK,0,0,0,0,
+    FUZZDARK,0,0,FUZZDARK,FUZZDARK,FUZZDARK,FUZZDARK,0,
+    FUZZDARK,0,FUZZDARK,FUZZDARK,0,0,FUZZDARK,
+    FUZZDARK,0,0,0,0,FUZZDARK,FUZZDARK,
+    FUZZDARK,FUZZDARK,0,FUZZDARK,FUZZDARK,0,FUZZDARK
+};
+
+static void DrawFuzzColumnRefraction(void)
+{
+    boolean cutoff = false;
 
-void R_SetFuzzColumnMode(void)
+    if (dc_x % fuzzblocksize)
+    {
+        return;
+    }
+
+    if (!dc_yl)
+    {
+        dc_yl = 1;
+    }
+
+    if (dc_yh == viewheight - 1)
+    {
+        dc_yh = viewheight - 2;
+        cutoff = true;
+    }
+
+    int count = dc_yh - dc_yl;
+
+    if (count < 0)
+    {
+        return;
+    }
+
+#ifdef RANGECHECK
+    if ((unsigned)dc_x >= video.width || dc_yl < 0 || dc_yh >= video.height)
+    {
+        I_Error("R_DrawFuzzColumn: %i to %i at %i", dc_yl, dc_yh, dc_x);
+    }
+#endif
+
+    ++count;
+
+    byte *dest = ylookup[dc_yl] + columnofs[dc_x];
+
+    int lines = fuzzblocksize - (dc_yl % fuzzblocksize);
+
+    int dark = FUZZDARK;
+    int offset = 0;
+
+    do
+    {
+        count -= lines;
+
+        // if (count < 0)
+        // {
+        //    lines += count;
+        //    count = 0;
+        // }
+        const int mask = count >> (8 * sizeof(mask) - 1);
+        lines += count & mask;
+        count &= ~mask;
+
+        const byte fuzz = fullcolormap[dark + dest[linesize * offset]];
+
+        do
+        {
+            memset(dest, fuzz, fuzzblocksize);
+            dest += linesize;
+        } while (--lines);
+
+        ++fuzzpos;
+
+        // Clamp table lookup index.
+        fuzzpos &= (fuzzpos - FUZZTABLE) >> (8 * sizeof(fuzzpos) - 1); // killough 1/99
+
+        dark = fuzzdark[fuzzpos];
+        offset = fuzzoffset[fuzzpos];
+
+        lines = fuzzblocksize;
+    } while (count);
+
+    if (cutoff)
+    {
+        const byte fuzz =
+            fullcolormap[dark + dest[linesize * (offset - FUZZOFF) / 2]];
+        memset(dest, fuzz, fuzzblocksize);
+    }
+}
+
+static void DrawFuzzColumnShadow(void)
```

Comment:
Hmm, "shadow" looks kind of rough to me. Is there any way to do this effect?

> I combined translucency (not grayscale) and selective (wavy effect) as an experiment for Crispy Strife a long time ago and thought it looked great. Just an idea.

https://github.com/user-attachments/assets/318a7451-3324-43e7-b9da-46f6f7a1081d

Reply:
The idea of "shadow" is to make sprites with "partial invisibility" more noticeable, it's an accessibility feature. I tried to draw a translucent sprite first and then apply the "refraction" effect, but it's still hard to see. Feel free to commit to this PR, maybe I did it wrong.

Reply:
The initial motivation for this was a user complaint about visibility issues with the original fuzz effect, so I am not sure if something like this qualifies as a solution to this request.